akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpAttributes.scala

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.engine.server
+
+import java.net.InetSocketAddress
+
+import akka.stream.Attributes
+
+/** Internal attributes used by the HTTP pipeline */
+object HttpAttributes {
+  import Attributes._
+
+  final case class RemoteAddress(address: Option[InetSocketAddress]) extends Attribute
+
+  def remoteAddress(address: Option[InetSocketAddress]) =
+    Attributes(RemoteAddress(address))
+
+}
```

Comment:
Yeah, I PRed early to see if this option is deemed ok to pursue at all or not; they'd be private then.